llvm/lib/Transforms/Instrumentation/HWAddressSanitizer.cpp

```diff
@@ -1532,8 +1539,8 @@ void HWAddressSanitizer::sanitizeFunction(Function &F,
 
   assert(!ShadowBase);
 
-  Instruction *InsertPt = &*F.getEntryBlock().begin();
-  IRBuilder<> EntryIRB(InsertPt);
+  BasicBlock::iterator InsertPt = F.getEntryBlock().begin();
```

Comment:
For my understanding: Can you quickly explain why this (and related changes below) are needed?

Reply:
We're in the process of changing how debug-info variable locations are recorded -- right now they're stored as intrinsics, but in the future we'll be able to attach them directly to instructions, which has a variety of benefits (efficiency and stability) because passes won't need to take account of instructions that aren't "real". As part of that, we need to put a tiny bit of debug-info into the instruction-iterator class, and have most instruction insertions be performed with iterators, hence this particular change on these lines.

I can go into detail, but by using iterators everywhere in scenarios like this, we're able to perform all the maintenance of debug-info in the background, without LLVM passes having to perform it themselves.

(There's a meaningful change that I'll describe in a main comment in a moment,)

Reply:
Thanks for the detailed explanation.